packages/@aws-cdk/custom-resource-handlers/test/aws-s3/notifications-resource-handler/Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM public.ecr.aws/lambda/python:3.7
+FROM public.ecr.aws/lambda/python:3.11
```

Comment:
I see integration tests were updated to use python3.11 previously here: https://github.com/aws/aws-cdk/pull/29506

For my understanding is this dockerfile is for running the integration tests? Also curious how there were no issues if it was using python3.7 before

Reply:
I will double check with the team but I believe we will also need to update the runtime in the  custom resource config file here: https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/custom-resource-handlers/lib/custom-resources-framework/config.ts#L265

Reply:
> For my understanding is this dockerfile is for running the integration tests?

This is for unit tests only AFAIK.

> Also curious how there were no issues if it was using python3.7 before

Python 3.7 and 3.11 are pretty much the same in this case. I am updating it in this PR in order to make unit test Lambda runtime to be the same version as in actual notification construct/Lambda.

Let me know if I need to update anything else. (Also, I am happy to revert this change if you think it's not needed in this PR)

Thanks for reviewing!

Reply:
Got it, thanks for the explanation. For the sake of parity (and to prevent future confusion), could you update the runtime in the custom-resources-framework `config.ts` file please? (will also need to add Python 3.11 enum [here](https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/custom-resource-handlers/lib/custom-resources-framework/config.ts#L7-L21))

Reply:
Done, thanks.